### PR TITLE
Ui 953 icon issues

### PR DESCRIPTION
**Pull request by sheela-svmx**
- Fix Icon Issues
- Clean up getIconColor file.

### CODE REVIEW

File: lib/components/Icon/IconContainer.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ const IconContainer = ({ ...props }) => {
   const { containerclassName, fillColor, shape, children } = props;
   let iconColor = null;
   React.Children.map(children, (child) => {
-    iconColor = getIconColor(fillColor, child.category, child.icon);
+    iconColor = getIconColor(fillColor, child.props.category, child.props.icon);
```
Comment (kokasvmx) [inline lib/components/Icon/IconContainer.js]: We should be doing a null check for child.props before retrieving category and icon

File: lib/components/Icon/getIconColor.js [hunk 2]
```diff
@@ -1,17 +1,14 @@
 const getIconColor = (fillColor, category, icon) => {
-  let iconColor = null;
+  if (fillColor) {
```
Comment (kokasvmx) [inline lib/components/Icon/getIconColor.js]: I see a functionality change from the actual file, Is this intended? What if the category is custom or action and fillcolor is not null? The actual will return based on category but the changed one will return fillcolor.

[the remark this answers is not in the document]
Comment (sheela-svmx) [reply lib/components/Icon/getIconColor.js]: So if the user Provides a fillcolor, then that will supercede and return that as the icon color. if the category is action and it contains the word "new_custom" then replace the iconColor with that string, else by default  return ```return `${category}-${(icon || '').replace(/_/g, '-')}`;```

Comment (sheela-svmx) [reply lib/components/Icon/getIconColor.js]: So previoulsy when i cleaned up this file,  there were a lot of places where null was returned, and it was not necessary and I had missed one check if the category was action that I added back that was causing the variable to be "undefined"

File: lib/components/Icon/IconContainer.js (hunk 1 above)
Comment (sheela-svmx) [reply lib/components/Icon/IconContainer.js]: @kokasvmx  we have default props set for these, so if there is no category or icon, the default prop will be selected

**Pull request merged**